src/plugin.js

```diff
@@ -60,9 +60,19 @@ module.exports = (config = {}) => {
 
     let muteFilter = (payload, next) => {
 
-        if(payload && payload.sender && payload.chat && payload.chat.muter.isMuted(payload.sender)) {
+        let isOwnEvent = false;
 
-            payload.chat.trigger('$.muter.eventRejected', payload);
+        if(payload && payload.event && payload.event.indexOf('$.muter') > -1) {
+            isOwnEvent = true;
+        }
+
+        if(!isOwnEvent && payload && payload.sender && payload.chat && payload.chat.muter && payload.chat.muter.isMuted(payload.sender)) {
```

Comment:
Thats a mighty big ```if()``` statement you got there.

Reply:
:*(

I know, it should be multiple variable assignment then a check for the result of those.